google-cloud-datastore/lib/google/cloud/datastore/transaction.rb

```diff
@@ -53,9 +55,7 @@ module Datastore
       #     tasks_in_list = tx.run query
       #   end
       #
-      class Transaction < Dataset
-        attr_reader :id
-
+      class Transaction < ReadOnlyTransaction
```

Comment:
Oops, I didn't realize that Transaction was inheriting from Dataset. Using inheritance here seems like a mistake, and something to move away from instead perpetuating. I know this is a larger issue than the scope of this PR, but I believe the use of inheritance here is likely going to cause more problems than it solves.

For instance, the `Dataset#transaction` and `Dataset#read_only_transaction` methods are going to be inherited. I believe inheritance can also make documentation more difficult than it would otherwise need to be. It would be cleaner to have separate `#run` methods on `Dataset`, `Transaction`, and `ReadOnlyTransaction` with code examples that were specific to the method's class, than having multiple code examples for all the ways the inherited method might be invoked.

I'd like to suggest we remove the use of inheritance, but I can see an argument for not doing that in this PR. I'd support a follow up PR for that. @quartzmo what do you think?

Reply:
Sounds good. I'll do it in this PR.